packages/react-dom/src/test-utils/ReactTestUtilsAct.js

```diff
@@ -119,6 +118,13 @@ function act(callback: () => Thenable) {
         called = true;
         result.then(
           () => {
+            if (actingUpdatesScopeDepth !== 1) {
```

Comment:
Is there any scenario where `actingUpdatesScopeDepth` would decrement twice, and this would be less than 1? Maybe should change to `actingUpdatesScopeDepth < 1` just in case.

Reply:
rather, actingUpdatesScopeDepth > 1